public/jobs.js

```diff
@@ -0,0 +1,477 @@
+//calling a function buildJobsTable.
+//This function draws the jobs table if there are any jobs.
+//This function is async because it will eventually await a fetch call to retrieve the list of jobs.
+// It returns the number of jobs retrieved.
+
+//here we use GET request for all of the jobs entries.
+//if no entries are returned, the function just returns 0. 
+//If entries are returned, they must be added to the table in the following columns: 
+//company, position, status, edit button, delete button. 
+//The rows of the table are accumulated in a loop, with the first row being the table header row.
+// The tricky part is the buttons. We need to identify whether a button represents an add
+// or delete. This is done with the editButton and deleteButton classes. 
+//We also have to record which jobs entry corresponds to which button. 
+//This is done with the dataset.id attribute, which is set in the HTML using dataset-id.
+// Then the HTML for each row is created and turned into a DOM entry. 
+//The table is updated with the rows using a replaceChildren() call.
+async function buildJobsTable(jobsTable, jobsTableHeader, token, message) {
+  try {
+    const response = await fetch('/api/v1/jobs', {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${token}`,
+      },
+    });
+    const data = await response.json();
+
+    var children = [jobsTableHeader]; //why jobsTableHeader, is that because tr is a parent element, can represent an array  ?
+    if (response.status === 200) {
+      if (data.count === 0) {
+        //  jobsTable.replaceChildren(...children); // clear this for safety
```

Comment:
@akosasante I did, but I didn´t notice any difference. what can I do to see the changes?

Reply:
Yeah nevermind I think in this case it doesn't actually make a difference; I'm not totally sure what the instructor intended here. Since the whole table gets hidden anyway if jobs returned is 0 :thinking: